microprofile.jdt/com.redhat.microprofile.jdt.core/src/main/java/com/redhat/microprofile/jdt/internal/core/ls/MicroProfileDelegateCommandHandler.java

```diff
@@ -127,6 +127,10 @@ private static MicroProfileProjectInfo getMicroProfileProjectInfo(List<Object> a
 		} catch (InterruptedException e) {
 			LOGGER.log(Level.WARNING, "Error while joining MicroProfile properties collector job", e);
 		}
+		
+		if (!job.getResult().isOK()) {
+			throw new UnsupportedOperationException(String.format("Cannot find MicroProfileProjectInfo for '%s'", params.getUri()));
```

Comment:
I would like to avoid throwing a new exception here since job could have anoter error. What about using `job.getResult().getException()` (or perhaps `job.getResult().getException().getCause()` to get the original error?

Reply:
Thank you, what do you think about the new code?

```
		Exception jobException = (Exception) job.getResult().getException();
		job.getResult().getException().getCause();

		if (jobException != null) {
			if (jobException.getCause() != null) {
				throw (Exception) jobException.getCause();
			}
			throw jobException;
		};
```

Reply:
yes it was my idea.